### DISCUSSION TITLE
有人来讨论效果和搭配组合么

### DISCUSSION THREAD

**Discussion by letterk** [Blinue/Magpie#364] [General]
1080p的显示器
为了提高帧率, 昨天测试了一整天
降低游戏内的分辨率, 再放大, 其实画面损失的不多, 但文字会变得很模糊. 尤其是一些游戏, 里面的UI字体可能是先渲染成图片再显示的, 一旦降低分辨率, 字体就成团了

许多算法很难看出有啥区别
目前我认为特点很鲜明的几个
ACNET, 除了2D平涂画风的动漫, 其他都不适合. 降噪强度已经大到磨皮的程度了.
LCAS, 好像是这两天才加入的, 这个算法的锐化是在边缘加一圈黑边, 测试了一圈, 发现只有这个算法, 性能消耗少, 还不会让文字变糊. 就是噪点特别多
Anime4K_Denoise_Bilateral_Mean.glsl, 这个降噪强度和ACNET差不多Mean,Mode,Median, 这三个我的理解就是针对噪点从AOE到精准打击的意思.
FSR这个算法是我找到magpie的原因, 更适合高分屏拿来做1080p -> 2k/4k这种. 放大不影响观感, 在不是高ppi的屏幕上, 会糊的一塌糊涂救都救不会来的那种, 比如我用的27寸1080p

**Comment by neavo** [comment]
> ![QQ截图20220511121917](https://user-images.githubusercontent.com/998676/167768341-76842ad3-dd78-44a1-8226-bc75e4f28f44.jpg)

> ![QQ截图20220511121927](https://user-images.githubusercontent.com/998676/167768346-654e51f8-febf-4168-b264-7103e3416ac1.jpg)

抛砖引玉，分享两个自用配置，对 低分辨率（720P↓） 和 中分辨率（1080P↑） 的ACG风格内容有奇效

**Reply by letterk** [reply]
720p和1080p分别用的是一次放大还是两次放大?
我看到你写了上下两个箭头, 没看明白?
你的屏幕是多少分辨率.

**Reply by neavo** [reply]
我是4K的显示设备，原分辨率720P以下的话用二次放大的配置较好，原分辨率1080P以上用一次放大的配置较好，可以实际试试，还是很明显得

**Reply by chaos700** [reply]
搭车问一下，有些视觉小说的立绘边缘的锯齿特别严重，
anime4k仍会保留这类锯齿，有什么算法适合减轻这个问题。

**Reply by neavo** [reply]
> 搭车问一下，有些视觉小说的立绘边缘的锯齿特别严重， anime4k仍会保留这类锯齿，有什么算法适合减轻这个问题。

试试 Anime4K 的 Restore，从我实测来看，在合适的步骤插入 Anime4K_Restore 是2DACG画面插值中最重要的一步，效果远大于放大插值算法本身

**Reply by letterk** [reply]
确实, 作为3D和真实影像放大, 不用AB对比, 根本看不出区别来. 2d上的线条用A4K重塑, 感知非常明显

**Reply by Mishini** [reply]
用原始数据2x，缩放到目标分辨率，最后再进行收线/锐化这样的流程比较好。收线/锐化提升观感的同时会损坏数据，这时候再放大缩小会将错误一并放大。唯一的问题就是原始数据越大，2x资源消耗越大

**Reply by neavo** [reply]
> 用原始数据2x，缩放到目标分辨率，最后再进行收线/锐化这样的流程比较好。收线/锐化提升观感的同时会损坏数据，这时候再放大缩小会将错误一并放大。唯一的问题就是原始数据越大，2x资源消耗越大

对4K分辨率进行Restore的性能开销太夸张了，9400F+1660S 大概只有30帧，而且锐度感觉不太够

配置够好的同学可以试试 先插值到目标分辨率最后再Restore + 锐化

**Reply by frdoma** [reply]
其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。

**Reply by letterk** [reply]
> 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。

RCAS这个锐化效果很迷, 默认0.87, 我看不出区别, 设置高一些反而更模糊. 降噪可以根据源单独选择三种Mean,Mode,Median.

**Reply by letterk** [reply]
一些老动画, 分辨率低, 渣压缩, 只能靠Restore和降噪来救一救, 单纯放大根本没法看

**Reply by Mishini** [reply]
> 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。 而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。 遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。

试了下效果不错，就是在fsr前面加了Anime4K_Upscale，然后用EASU缩放RCAS锐化

**Reply by frdoma** [reply]
> > 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。
> 
> RCAS这个锐化效果很迷, 默认0.87, 我看不出区别, 设置高一些反而更模糊. 降噪可以根据源单独选择三种Mean,Mode,Median.

RCAS好像最多到1.0就可以了，再多会出现噪点，如果嫌锐化效果不明显就用CAS，最多也是到1.0但是比RCAS强力很多。个人测试很多之后感觉大部分情况下用0.87或用更低的RCAS就可以了，锐化过高的话锯齿感变重、画面也会变得不自然。

**Reply by neavo** [reply]
> 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。

![Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen](https://user-images.githubusercontent.com/998676/168196190-13cc5e98-3f4a-4d0b-a7d1-85b7b9c90419.jpg)

![Anime4K_Upscale_L + FSR](https://user-images.githubusercontent.com/998676/168196192-61005f49-2986-472d-95d4-83331aa8a684.jpg)

先上结论：在4K显示器上，低分辨率2DACG内容，Anime4K_Upscale_L + FSR 还是不太行

上图是 Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen 
下图是 Anime4K_Upscale_L + FSR_EASU + FSR_RCAS

缩小以后的截图看似相差无几，但是请将图片文件到本地对比查看，如果你在使用4K的屏幕的话，效果差别是显而易见的

虽然加入 Anime4K_Restore 的配置确实有很明显的涂抹感（主要是文字部分），但是整体观感还是要明显好于对比组的
在性能上两者也相差无几，处理时间都在12ms左右

如果你得出了跟我相反的结论，是否是因为你的图像源分辨率很高或者屏幕分辨率很低 ？

**Reply by neavo** [reply]
> > 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。
> 
> ![Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen](https://user-images.githubusercontent.com/998676/168196190-13cc5e98-3f4a-4d0b-a7d1-85b7b9c90419.jpg)
> 
> ![Anime4K_Upscale_L + FSR](https://user-images.githubusercontent.com/998676/168196192-61005f49-2986-472d-95d4-83331aa8a684.jpg)
> 
> 先上结论：低分辨率2DACG内容，Anime4K_Upscale_L + FSR 还是不太行
> 
> 上图是 Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen 下图是 Anime4K_Upscale_L + FSR_EASU + FSR_RCAS
> 
> 缩小以后的截图看似相差无几，但是请将图片文件到本地对比查看，如果你在使用4K的屏幕的话，效果差别是显而易见的
> 
> 虽然加入 Anime4K_Restore 的配置确实有很明显的涂抹感（主要是文字部分），但是整体观感还是要明显好于对比组的 在性能上两者也相差无几，处理时间都在12ms左右
> 
> 如果你得出了跟我相反的结论，是否是因为你的图像源分辨率很高或者屏幕分辨率很低 ？

两组的具体配置如下：

```
[
	{
		"name": "Anime4K + FSR",
		"effects": [
			{
				"effect": "Anime4K_Upscale_L"
			},
			{
				"effect": "FSR_EASU",
				"scale": [
					-1,
					-1
				]
			},
			{
				"effect": "FSR_RCAS",
				"sharpness": 0.87
			}
		]
	},
	{
		"name": "2D - 一次放大",
		"effects": [
			{
				"effect": "Jinc",
				"scale": [
					-0.5,
					-0.5
				]
			},
			{
				"effect": "Anime4K_Restore_L"
			},
			{
				"effect": "Anime4K_Thin_HQ"
			},
			{
				"effect": "Anime4K_Upscale_L"
			},
			{
				"effect": "AdaptiveSharpen"
			}
		]
	},
]
```

**Reply by Mishini** [reply]
> > 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。
> 
> ![Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen](https://user-images.githubusercontent.com/998676/168196190-13cc5e98-3f4a-4d0b-a7d1-85b7b9c90419.jpg)
> 
> ![Anime4K_Upscale_L + FSR](https://user-images.githubusercontent.com/998676/168196192-61005f49-2986-472d-95d4-83331aa8a684.jpg)
> 
> 先上结论：低分辨率2DACG内容，Anime4K_Upscale_L + FSR 还是不太行
> 
> 上图是 Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen 下图是 Anime4K_Upscale_L + FSR_EASU + FSR_RCAS
> 
> 缩小以后的截图看似相差无几，但是请将图片文件到本地对比查看，如果你在使用4K的屏幕的话，效果差别是显而易见的
> 
> 虽然加入 Anime4K_Restore 的配置确实有很明显的涂抹感（主要是文字部分），但是整体观感还是要明显好于对比组的 在性能上两者也相差无几，处理时间都在12ms左右
> 
> 如果你得出了跟我相反的结论，是否是因为你的图像源分辨率很高或者屏幕分辨率很低 ？

场景不同了，A4k+fsr，1080fsr是缩小，4k就是放大了
我就1.4x，所以2x再缩回来挺好的。大于2x可以走2x接2x，但是4k屏后处理放最后吃不消(有些游戏可以不用满帧)，0.5x-后处理-2x挺好的。1080p遇到2x接2x的概率不大。有兴趣的话可以试下2x接2x再fsr，性能应该比不了，画质你描述一下就行了，我相信你说的

**Reply by frdoma** [reply]
> > > 其实Anime4k+FSR+RCAS就可以了，Restore对显卡负载过高，而且更要命的是容易出现明显的涂抹感，不够自然。而GAN只是实验性阶段，有过锐化、细节显脏和明显振铃的问题，所以我感觉Anime4k无降噪版就好（降噪版在降噪的同时也会抹掉细节，看个人取舍）。遗憾的是目前Magpie里Anime4k最好的也只是L，希望未来能移植UL和VL来获得更好的效果。
> > 
> > 
> > ![Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen](https://user-images.githubusercontent.com/998676/168196190-13cc5e98-3f4a-4d0b-a7d1-85b7b9c90419.jpg)
> > ![Anime4K_Upscale_L + FSR](https://user-images.githubusercontent.com/998676/168196192-61005f49-2986-472d-95d4-83331aa8a684.jpg)
> > 先上结论：低分辨率2DACG内容，Anime4K_Upscale_L + FSR 还是不太行
> > 上图是 Anime4K_Restore_L + Anime4K_Thin_HQ + Anime4K_Upscale_L + AdaptiveSharpen 下图是 Anime4K_Upscale_L + FSR_EASU + FSR_RCAS
> > 缩小以后的截图看似相差无几，但是请将图片文件到本地对比查看，如果你在使用4K的屏幕的话，效果差别是显而易见的
> > 虽然加入 Anime4K_Restore 的配置确实有很明显的涂抹感（主要是文字部分），但是整体观感还是要明显好于对比组的 在性能上两者也相差无几，处理时间都在12ms左右
> > 如果你得出了跟我相反的结论，是否是因为你的图像源分辨率很高或者屏幕分辨率很低 ？
> 
> 两组的具体配置如下：
> 
> ```
> [
> 	{
> 		"name": "Anime4K + FSR",
> 		"effects": [
> 			{
> 				"effect": "Anime4K_Upscale_L"
> 			},
> 			{
> 				"effect": "FSR_EASU",
> 				"scale": [
> 					-1,
> 					-1
> 				]
> 			},
> 			{
> 				"effect": "FSR_RCAS",
> 				"sharpness": 0.87
> 			}
> 		]
> 	},
> 	{
> 		"name": "2D - 一次放大",
> 		"effects": [
> 			{
> 				"effect": "Jinc",
> 				"scale": [
> 					-0.5,
> 					-0.5
> 				]
> 			},
> 			{
> 				"effect": "Anime4K_Restore_L"
> 			},
> 			{
> 				"effect": "Anime4K_Thin_HQ"
> 			},
> 			{
> 				"effect": "Anime4K_Upscale_L"
> 			},
> 			{
> 				"effect": "AdaptiveSharpen"
> 			}
> 		]
> 	},
> ]
> ```

4k分辨率下用这个试了一下确实如你所说的，单纯靠Anime4K+FSR甚至Anime4K x2清晰度都不行，必须靠Restore清晰度才能提升很多，只能用涂抹感换取清晰度。为追求尽量自然我把Restore L换成Restore Soft L，锐化换成RCAS并且放在FSR之前，这是Restore Soft L+Anime4k L+RCAS+FSR的效果图，相对柔和自然一些。
![Restore Soft L+Anime4k L+RCAS+FSR](https://user-images.githubusercontent.com/86701052/168224349-c04172d6-6ef7-4702-8659-08e9ce6984c8.png)

目前ADV和同人RPG的主流分辨率还是720p，一般放大2倍之内效果还相对能接受，所以屏幕最多2k就可以了，4k屏真是徒增烦恼（不过可以试试整数倍缩放把屏幕分辨率设置成1080p？）。
这是Anime4K+Lanczos在1080p和1440p下的效果图，分辨率越高效果越差是没办法的事（FSR貌似并不是被设计来下采样的，只有上采样的时候才有必要用，所里这里换成Lanczos）。
![1080p](https://user-images.githubusercontent.com/86701052/168221551-fa39eaae-501d-472a-b43c-4183c99e0313.png)
![1440p](https://user-images.githubusercontent.com/86701052/168221569-c32bd566-3749-42f6-b46c-38d5f0ed1ff3.png)

**Comment by Mishini** [comment]
参考madvr和mpv的视频播放教程，1.2倍的用Jinc，大于1.5小于等于2倍的用nnedi3+Bicubic

**Reply by letterk** [reply]
那有视频播放教程?

**Reply by neavo** [reply]
至少对于2DACG风格的画面来说，普通的缩放算法实际区别不大，关键还是得靠A4K脑补 。。。。
